drivers/dma/dma_esp32_gdma.c

```diff
@@ -211,20 +241,45 @@ static int dma_esp32_config_rx(const struct device *dev, struct dma_esp32_channe
 }
 
 static int dma_esp32_config_tx_descriptor(struct dma_esp32_channel *dma_channel,
-					   struct dma_block_config *block)
+						struct dma_block_config *block)
 {
 	if (!esp_ptr_dma_capable((uint32_t *)block->source_address)) {
 		LOG_ERR("Tx buffer not in DMA capable memory: %p",
 			(uint32_t *)block->source_address);
 		return -EINVAL;
 	}
 
-	memset(&dma_channel->desc, 0, sizeof(dma_channel->desc));
-	dma_channel->desc.buffer = (void *)block->source_address;
-	dma_channel->desc.dw0.size = block->block_size;
-	dma_channel->desc.dw0.length = block->block_size;
-	dma_channel->desc.dw0.suc_eof = 1;
-	dma_channel->desc.dw0.owner = DMA_DESCRIPTOR_BUFFER_OWNER_DMA;
+	if (!block) {
```

Comment:
This block could go as the first in this function as well, to protect 'block' access